astrokat/noisediode.py

```diff
@@ -175,6 +175,34 @@ def _nd_log_msg_(ant,
     return actual_time
 
 
+def nd_reset(kat,
+             timestamp,
+             switch=0):
+    """Reset noise-source on or off.
+
+    Parameters
+    ----------
+    kat : session kat container-like object
+        Container for accessing KATCP resources allocated to schedule block.
+    timestamp : float, optional
+        Time since the epoch as a floating point number [sec]
+    switch: int, optional
+        off = 0 (default), on = 1
+
+    Returns
+    -------
+    timestamp : float
+        Linux timestamp reported by digitiser
+    """
+    antennas = sorted(ant.name for ant in kat.ants)
+    user_logger.info('Resetting all noise diodes')
+    # TODO - bngcebetsha: do we want to do this for ALL antennas or a targeted group such
+    # as is done in `_set_dig_nd_()`?
+    for ant in antennas:
```

Comment:
There is no need to loop over antennas... Use the `kat.ants` group instead. I.e. `kat.ants.req.dig_noise_source(timestamp, switch)`.